src/components/settings/members/CreateInviteDialog.tsx

```diff
@@ -0,0 +1,193 @@
+
+export const CreateInviteDialog = forwardRef<DialogRef>((_, ref) => {
+  const { translate } = useInternationalization()
+  const { currentOrganization } = useCurrentUserInfosVar()
+  const [inviteToken, setInviteToken] = useState<string>()
+  const [createInvite, { error }] = useCreateInviteMutation({
+    onCompleted(res) {
+      if (res?.createInvite?.token) {
+        setInviteToken(res.createInvite.token)
+      }
+    },
+  })
+
+  const formikProps = useFormik<CreateInviteInput>({
+    initialValues: {
+      email: '',
+    },
+    validationSchema: object().shape({
+      email: string().email().required(''),
+    }),
+    enableReinitialize: true,
+    validateOnMount: true,
+    onSubmit: async (values, formikBag) => {
+      const result = await createInvite({
+        variables: {
+          input: {
+            ...values,
+          },
+        },
+        refetchQueries: ['getInvites'],
+      })
+
+      const { errors } = result
+
+      const apiError = !errors ? undefined : (errors[0]?.extensions as LagoGQLError['extensions'])
+
+      if (
+        !!apiError &&
+        apiError?.code === Lago_Api_Error.UnprocessableEntity &&
+        !!apiError?.details?.invite
+      ) {
+        formikBag.setFieldError('email', translate('text_63208c701ce25db781407456'))
+      }
+    },
+  })
+
+  return (
+    <Dialog
+      ref={ref}
+      title={translate('text_63208c701ce25db78140748f')}
+      description={
+        inviteToken
+          ? translate('text_63208c701ce25db78140743a', {
+              organizationName: currentOrganization?.name,
+            })
+          : translate('text_63208c701ce25db78140749b')
+      }
+      onOpen={() => {
+        formikProps.validateForm()
+      }}
+      onClickAway={() => {
+        setInviteToken('')
+        formikProps.resetForm()
+        formikProps.validateForm()
+      }}
+      actions={({ closeDialog }) => (
+        <>
+          <Button
+            variant="quaternary"
+            onClick={() => {
+              setInviteToken('')
+              closeDialog()
+              formikProps.resetForm()
+              formikProps.validateForm()
+            }}
+          >
+            {translate('text_63208c711ce25db7814074cd')}
+          </Button>
+          {!inviteToken ? (
+            <Button
+              disabled={!formikProps.isValid}
+              onClick={async () => {
+                await formikProps.submitForm()
+              }}
+            >
+              {translate('text_63208c711ce25db7814074d9')}
+            </Button>
+          ) : (
+            <Button
+              disabled={!!error}
+              onClick={() => {
+                navigator.clipboard.writeText(`${window.location.origin}/invitation/${inviteToken}`)
+                addToast({
+                  severity: 'info',
+                  translateKey: 'text_63208c711ce25db781407536',
+                })
+                setInviteToken('')
```

Comment:
You're doing all this a bunch of times, might be a good idea to make a function out of it